### PR TITLE
[TypeSpec CI] Add CI and PR triggers

### PR DESCRIPTION
**Pull request by mikeharder**

### CODE REVIEW

File: eng/pipelines/typespec-ci.yml [hunk 1]
```diff
@@ -1,3 +1,19 @@
+trigger:
+  branches:
+    include:
+    - main
```
Comment (weshaggard) [inline eng/pipelines/typespec-ci.yml:4]: So you want this run CI on every commit into main? I think that is what this is saying.

**Review by weshaggard — APPROVED** [review]
Looks good other than my question about CI.

File: eng/pipelines/typespec-ci.yml (hunk 1 above)
Comment (mikeharder) [reply eng/pipelines/typespec-ci.yml:4]: The pipeline only takes 1-2 minutes so it seemed reasonable to run on every commit to main.  But I also don't mind only running this on a nightly schedule, and either no CI trigger, or scoping the CI trigger to the same paths as the PR trigger.

Comment (mikeharder) [reply eng/pipelines/typespec-ci.yml:4]: @weshaggard: Would you prefer this pipeline not run on every commit to main, or were you just asking if it was intentional?

Comment (weshaggard) [reply eng/pipelines/typespec-ci.yml:4]: I'm OK with having it run on every commit for now and if there are issues we can tune it.

**Pull request merged**